composer.json

```diff
@@ -38,6 +38,7 @@
     },
     "require": {
         "php": ">=5.3",
+        "ext-sockets": "*",
```

Comment:
Depending how we look at it, this could be a BC break

Reply:
Agree that this is problematic. We could make this an optional suggestion, but in this case the entire API surface would be less predictable. Not sure what the best course of action would be here, perhaps we can expose the stream resource somehow to give more control to consumers that need low-level socket settings? (Also refs https://github.com/reactphp/socket/issues/150)

Reply:
just a question: is the socket-extension not part of the default RTL? so we do not have this problem anyway?

Reply:
`ext-sockets` is not compiled by default as per https://www.php.net/manual/en/sockets.installation.php, but it is commonly included as part of many distributions.

Reply:
i think this problem will never happen, so as you already have said:
> commonly included as part of many distributions.

Reply:
I understand it's compelling to consider this to be irrelevant, but we've seen numerous reports of non-standard installations, most recently https://github.com/leproxy/leproxy/issues/80 (and linked tickets). Unfortunately "commonly" does not equal "always".

If we *require* this extension, then we need to add it to `composer.json`. This in turn could be considered as BC break, as platforms that do not currently have this extension can no longer upgrade.

I understand and agree this can be annoying, but I'd rather be conservative here as we have to keep in mind the effect on the larger ecosystem.

Reply:
the code is safe, so it handle a missing socket-extension and ignore this new feature... maybe we should trigger and info/error to inform to this behavior.
just remove this require from the composer.json? and add an info to the api-doku?

Reply:
> the code is safe

Yeah, that is well done, was really happy to see that 👍 . It's just a comment that adding it there would be a BC break. And I think moving it to suggest + adding docs is the best way forward here.